src/BackupDestination/Backup.php

```diff
@@ -27,24 +36,38 @@ public function path(): string
 
     public function exists(): bool
     {
-        return $this->disk->exists($this->path);
+        if ($this->exists === null) {
+            $this->exists = $this->disk->exists($this->path);
+        }
+
+        return $this->exists;
     }
 
     public function date(): Carbon
     {
-        return Carbon::createFromTimestamp($this->disk->lastModified($this->path));
+        if ($this->date === null) {
+            $this->date = Carbon::createFromTimestamp($this->disk->lastModified($this->path));
+            $this->exists = true;
+        }
+
+        return $this->date;
     }
 
     /**
      * Get the size in bytes.
      */
     public function size(): int
     {
-        if (! $this->exists()) {
-            return 0;
+        if ($this->size === null) {
+            if (! $this->exists()) {
```

Comment:
Do we really need to set `exists` here? This code would be much simpler if we only think of the  `size` here.